lib/k8s/operation.ex

```diff
@@ -20,14 +23,22 @@ defmodule K8s.Operation do
     apply: :patch
   }
 
+  @patch_type_header_map %{
+    merge: ["Content-Type": "application/merge-patch+json"],
+    strategic_merge: ["Content-Type": "application/strategic-merge-patch+json"],
+    json_merge: ["Content-Type": "application/json-patch+json"],
+    apply: ["Content-Type": "application/apply-patch+yaml"]
+  }
+
   defstruct method: nil,
             verb: nil,
             api_version: nil,
             name: nil,
             data: nil,
             conn: nil,
             path_params: [],
-            query_params: []
+            query_params: [],
+            header_params: []
```

Comment:
How about we define a default here so we can safely remove the code in `base.ex`? Just a thought. Would have to run tests.

```suggestion
            header_params: ["Content-Type": "application/json"]
```

Reply:
I did add this, and `mix test` is succeeding.  I do think there is a case, where if an application was creating the Operation struct themselves, without using build, where this would be a breaking change if the Operation was a patch or apply, since it was the Client.Runner.Base that was setting these headers before.  That seems like it would be really rare though?